### ISSUE TITLE
Memory usage on on checkout/smudge

### ISSUE THREAD

**Issue by joerg** [github/git-lfs#448]
I am playing around with my own Java Implementation of git-lfs Server (will release it in max 3 weeks time) and found that on checkout the full file gets stuffed into RAM. I was testing around with some 3 GB files and did a streaming upload and download. I first thought that my JAVA implementation of the server was broken, but running "git clone" from a different PC also results in huge memory spike upon checkout. The download works just fine and is not memory consuming, just the final checkout seems to be intensive.
Can anyone confirm this using the reference server implementation using any 3 to 4 GB file?

**Comment by whoisj** [comment]
Which version of Git are you using? Git Cli should be streaming if I'm not mistaken.

**Comment by joerg** [comment]
Hi, I tried monitoring the memory usage and it turns out, that this is not the fault of git-lfs. I did a rather simple plot using "ps -C git-lfs -o pid=,%mem=,vsz=" and it stayed low all the time. But with a large file (3GB) I was able to see that git-read-tree was the problem. At least at version 2.1.0 (default on Ubuntu 14.10) it reads the whole file into RAM.
I am gonna close this, but I think sooner or later this will come back to being an git-lfs problem...
